### PR TITLE
Add initial support for TypeVarLike default parameters (PEP 696)

### PR DESCRIPTION
**Pull request by cdce8p**
This adds the `default` parameter to `TypeVar`, `ParamSpec`, and `TypeVarTuple`.
PR: https://github.com/python/typing_extensions/pull/77

https://peps.python.org/pep-0696/

/CC: @Gobot1234, @JelleZijlstra

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Thanks! We'll need to wait for the runtime PR to be merged before we can merge this :)

File: stdlib/typing_extensions.pyi [hunk 1]
```diff
@@ -268,3 +249,56 @@ else:
...
+    __contravariant__: bool
+    __default__: Any | None
+    def __init__(
+        self,
+        name: str,
+        *constraints: Any,
+        bound: Any | None = ...,
+        covariant: bool = ...,
+        contravariant: bool = ...,
+        default: Any | None = ...,
```
Comment (JelleZijlstra) [inline stdlib/typing_extensions.pyi:270]:
```suggestion
        default: Any | None = ...,
        infer_variance: bool = ...,
```
Might as well add this one too.

File: stdlib/typing_extensions.pyi [hunk 2]
```diff
@@ -268,3 +249,56 @@ else:
             def _asdict(self) -> collections.OrderedDict[str, Any]: ...
 
         def _replace(self: _typeshed.Self, **kwargs: Any) -> _typeshed.Self: ...
+
+# New things in 3.xx
+# The `default` parameter was added to TypeVar, ParamSpec, and TypeVarTuple (PEP 696)
```
Comment (JelleZijlstra) [inline stdlib/typing_extensions.pyi:254]:
```suggestion
# The `default` parameter was added to TypeVar, ParamSpec, and TypeVarTuple (PEP 696)
# The `infer_variance` parameter was added to TypeVar (PEP 695)
```

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Actually it looks like we need to add a few more things to get CI green with the new release. I'll just hijack this PR to also add override and Any.

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**